kore/src/Kore/Simplify/AndTerms.hs

```diff
@@ -613,42 +636,47 @@ unifySortInjection termMerger term1 term2 unifyInj = do
         let (childTerm, childCondition) = Pattern.splitTerm childPattern
             inj' = evaluateInj inj{injChild = childTerm}
         return $ Pattern.withCondition inj' childCondition
+    UnifyInjData{unifyInj, term1, term2} = unifyData
+
+data ConstructorSortInjectionAndEquals = ConstructorSortInjectionAndEquals
+    { term1, term2 :: !(TermLike RewritingVariableName)
+    }
 
 {- | Matches
 
 @
-\\equals{_, _}(inj{_,_}(_), f(_))
+\\equals{_, _}(inj{_,_}(_), c(_))
 @
 
 @
-\\equals{_, _}(f(_), inj{_,_}(_))
+\\equals{_, _}(c(_), inj{_,_}(_))
 @
 
 and
 
 @
-\\and{_}(inj{_,_}(_), f(_))
+\\and{_}(inj{_,_}(_), c(_))
 @
 
 @
-\\and{_}(f(_), inj{_,_}(_))
+\\and{_}(c(_), inj{_,_}(_))
 @
 
-when @f@ has the @constructor@ attribute.
+when @c@ has the @constructor@ attribute.
 -}
 matchConstructorSortInjectionAndEquals ::
     TermLike RewritingVariableName ->
     TermLike RewritingVariableName ->
-    Maybe ()
+    Maybe ConstructorSortInjectionAndEquals
 matchConstructorSortInjectionAndEquals first second
     | Inj_ _ <- first
       , App_ symbol _ <- second
       , Symbol.isConstructor symbol =
-        Just ()
+        Just ConstructorSortInjectionAndEquals{term1 = first, term2 = second}
     | Inj_ _ <- second
       , App_ symbol _ <- first
       , Symbol.isConstructor symbol =
-        Just ()
+        Just ConstructorSortInjectionAndEquals{term1 = first, term2 = second}
```

Comment:
Does the `term1` and `term2` distinction not matter? I'm asking because in both cases `term1 = first` and `term2 = second`.

Reply:
Yes, currently the code for `constructorSortInjectionAndEquals` does not change the order of the two terms depending on which matches with the constructor versus the injection.